### ISSUE TITLE
Incomplete subnet list on one node (3-node setup)

### ISSUE THREAD

**Issue by micw** [coreos/flannel#1237]
I have a 3 node cluster build with rancher2 with "canal" network which uses flannel v0.11.0 for overlay networking.  That works on several clusters with absolutely no issues.
On one cluster we replaced the 3 nodes by adding 3 new and then removing the 3 old nodes. After that we had stange network issues which we tracked down to flannel which has one of the routes missing on one of the nodes.

## Expected Behavior
Flannel should add 2 routes to the other nodes on each node.

## Current Behavior
On one node it adds only one route

## Possible Solution
Unknown.

## Steps to Reproduce (for bugs)
Unknown. I have access to this setup until Jan/6 2020 and can provide as much info as possible to debug it.

## Context
The issue avoids routing of traffic from that particular node to pods on the other node.
We have completely removed kubernetes (all docker containers services, data) from the faulty node, reinstalled everything, re-added to the cluster - same result.

## Your Environment
* Flannel version: v0.11.0
* Backend used: vxlan
* Etcd version: rancher/coreos-etcd:v3.3.15-rancher1
* Kubernetes version (if used): v1.16.3
* Operating System and version: debian 10

**Comment by micw** [comment]
Log from the faulty node:
```
I1219 13:52:07.463031       1 vxlan_network.go:60] watching for new subnet leases
I1219 13:52:07.463166       1 vxlan_network.go:138] adding subnet: 10.42.5.0/24 PublicIP: 172.16.0.6 VtepMAC: 96:64:75:86:a1:56
I1219 13:52:07.463199       1 device.go:156] calling AddARP: 10.42.5.0, 96:64:75:86:a1:56
I1219 13:52:07.463310       1 device.go:133] calling AddFDB: 172.16.0.6, 96:64:75:86:a1:56
```

**Comment by micw** [comment]
I have enabled full debug (-v=10). I can see that all 3 nodes query the cluster nodes using kubectl and all 3 nodes can see all nodes in the list (including their correct pod ip ranges).

**Comment by micw** [comment]
I have figured out that not only the route is missing. When I add the route manually, it still does not work.

**Comment by micw** [comment]
Can anyone give a hint how I can debug this?

**Comment by micw** [comment]
Got some success: I went through the code and saw that "flannel.alpha.coreos.com/public-ip" annotaion is expected to get the subnet created. That was missing on one node. After I restarted flannel on that node, the annotation was added, resulting in creation of the missing route on the other node.

Strange thing is that I had rebooted all nodes before which did not fix this.

I'll now reboot all 3 nodes and check the result.
